From 922e85909b7d91deeb48213453ad4f1aaa9f3712 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:06:30 -0400
Subject: [powerpc] Fix PPC_EMULATED_STATS build break with sync patch

Message-id: <1410545655-205645-162-git-send-email-dzickus@redhat.com>
Patchwork-id: 94175
O-Subject: [RHEL7 PATCH 161/626] powerpc: Fix PPC_EMULATED_STATS build break with sync patch
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit a3821b2af185b64e3382c45fbdaa2cbc91ce14b8
Author: Scott Wood <scottwood@freescale.com>
Date:   Mon Oct 28 22:07:59 2013 -0500

    powerpc: Fix PPC_EMULATED_STATS build break with sync patch

    Commit 9863c28a2af90a56c088f5f6288d7f6d2c923c14 ("powerpc: Emulate sync
    instruction variants") introduced a build breakage with
    CONFIG_PPC_EMULATED_STATS enabled.

    Signed-off-by: Scott Wood <scottwood@freescale.com>
    Cc: Kumar Gala <galak@kernel.org>
    Cc: James Yang <James.Yang@freescale.com>
    ---
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/asm/emulated_ops.h b/arch/powerpc/include/asm/emulated_ops.h
index 63f2a22..f19b847 100644
--- a/arch/powerpc/include/asm/emulated_ops.h
+++ b/arch/powerpc/include/asm/emulated_ops.h
@@ -43,6 +43,7 @@ extern struct ppc_emulated {
  struct ppc_emulated_entry popcntb;
  struct ppc_emulated_entry spe;
  struct ppc_emulated_entry string;
+ struct ppc_emulated_entry sync;
  struct ppc_emulated_entry unaligned;
 #ifdef CONFIG_MATH_EMULATION
  struct ppc_emulated_entry math;
diff --git a/arch/powerpc/kernel/traps.c b/arch/powerpc/kernel/traps.c
index 4192de9..a289711 100644
--- a/arch/powerpc/kernel/traps.c
+++ b/arch/powerpc/kernel/traps.c
@@ -1866,6 +1866,7 @@ struct ppc_emulated ppc_emulated = {
  WARN_EMULATED_SETUP(popcntb),
  WARN_EMULATED_SETUP(spe),
  WARN_EMULATED_SETUP(string),
+ WARN_EMULATED_SETUP(sync),
  WARN_EMULATED_SETUP(unaligned),
 #ifdef CONFIG_MATH_EMULATION
  WARN_EMULATED_SETUP(math),
-- 
1.7.1